pkg/capabilities/consensus/ocr3/capability.go

```diff
@@ -109,6 +114,9 @@ func (o *capability) HealthReport() map[string]error {
 }
 
 func (o *capability) RegisterToWorkflow(ctx context.Context, request capabilities.RegisterToWorkflowRequest) error {
+	o.registeredWorkflowsMu.Lock()
+	defer o.registeredWorkflowsMu.Unlock()
```

Comment:
For more cohesion you could move these two lines just above where it's used; that also minimises the amount of time the lock is held to the minimum necessary